python/GafferSceneUI/SceneViewUI.py

```diff
@@ -1073,13 +1076,105 @@ def __appendClippingPlaneMenuItems( menuDefinition, prefix, view, parentWidget )
 			}
 		)
 
+def __snapshotDescription( view ) :
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	if sceneGadget.getRenderer() == "OpenGL" :
+		return "Viewport snapshots are only available for rendered (non-OpenGL) previews."
+
+	return "Snapshot viewport and send to catalogue."
+
+def __snapshotToCatalogue( catalogue, view ) :
+
+	timeStamp = str( datetime.datetime.now() )
+
+	fileName = pathlib.Path(
+		view["in"].node().getContext().substitute( catalogue["directory"].getValue() )
+	) / ( timeStamp.replace( ":", "-" ).replace(" ", "_" ) + ".exr" )
+
+	resolutionGate = imath.Box3f()
+	if isinstance( view, GafferSceneUI.SceneView ) :
+		resolutionGate = view.resolutionGate()
+
+	scriptRoot = view["in"].getInput().ancestor( Gaffer.ScriptNode )
+	metadata = IECore.CompoundData(
+		{
+			"gaffer:sourceScene" : view["in"].getInput().relativeName( scriptRoot ),
+			"gaffer:context:frame" : view["in"].node().getContext().getFrame()
+		}
+	)
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	sceneGadget.snapshotToFile( fileName, resolutionGate, metadata )
+
+	image = GafferImage.Catalogue.Image( "Snapshot1", Gaffer.Plug.Direction.In, Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+	image["fileName"].setValue( fileName )
+
+	image["description"].setValue(
+		"Snapshot of {} at frame {} taken at {}".format(
+			metadata["gaffer:sourceScene"],
+			metadata["gaffer:context:frame"], timeStamp[:-6]  # Remove trailing microseconds
+		)
+	)
+
+	catalogue["images"].source().addChild( image )
+
+def __snapshotCataloguesSubMenu( view, scriptNode ) :
+
+	menuDefinition = IECore.MenuDefinition()
+
+	catalogueList = list( GafferImage.Catalogue.RecursiveRange( scriptNode ) )
+
+	if len( catalogueList ) == 0 :
+		menuDefinition.append(
+			"/No Catalogues Available",
+			{
+				"active" : False,
+			}
+		)
+
+	else :
+		snapshotDescription = __snapshotDescription( view )
+		snapshotActive = view["renderer"]["name"].getValue() != "OpenGL"
+
+		for c in catalogueList :
+
+			cName = c["name"].getValue()
+			nName = c.relativeName( scriptNode )
```

Comment:
When a Catalogue has been put in a Box, Reference or custom node and then promoted (which is very common), this is giving the name of the internal Catalogue node which might well be private. I think we want to use the name of the promoted-to node in this case, which can be done pretty easily with `catalogue["imageIndex"].source().node()`.